cmd/vic-init/main_linux.go

```diff
@@ -201,3 +254,28 @@ func defaultIP() string {
 
 	return toolbox.DefaultIP()
 }
+
+func startSignalHandler() {
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGHUP, syscall.SIGUSR1, syscall.SIGUSR2, syscall.SIGPWR, syscall.SIGTERM, syscall.SIGINT)
+
+	go func() {
+		for s := range sigs {
+			switch s {
+			case syscall.SIGHUP:
+				log.Infof("Reloading tether configuration")
+				tthr.Reload()
+			case syscall.SIGUSR1, syscall.SIGUSR2, syscall.SIGPWR:
+				log.Infof("Stopping tether via signal %s", s.String())
+				tthr.Stop()
```

Comment:
don't you want to add return statement after ttht.Stop() to make sure goroutine finishes it's work?

Reply:
`tether.Stop` should be the end of things. We probably want to allow this behaviour to be configurable at some point, but if we've not shutdown as part of this operation I'd rather leave the signal handling running.